### PR TITLE
New/gum 59 initial password setup

### PR DESCRIPTION
**Pull request by jol79**

### CODE REVIEW

File: src/screens/complete-registration/complete-registration.tsx [hunk 1]
```diff
@@ -1,19 +1,31 @@
...
 
-  return <Text>CompleteRegistration screen with token {token}!</Text>;
+  return (
+    <>
+      <SafeAreaView style={styles.screen}>
+        <View style={styles.content}>
+          <CompleteRegistrationForm accessToken={token} />
+        </View>
+      </SafeAreaView>
+    </>
```
Comment (liza-veis) [inline src/screens/complete-registration/complete-registration.tsx]: Wrap it in AuthScreen and remove styles.ts file as it done for RegisterScreen and LoginScreen

File: src/screens/complete-registration/components/complete-registration-form/complete-registration-form.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import { View } from 'react-native';
+import { Formik } from 'formik';
+
+import { completeRegistrationValidationSchema } from 'src/validation-schemas';
+import { FormPasswordInput, MainButton } from 'src/components';
+import { ICompleteRegistrationPayload } from 'src/common/types';
+import { AuthRoute } from 'src/common/enums';
+
+import { useAppDispatch, useAppNavigation } from 'src/hooks';
```
Comment (liza-veis) [inline src/screens/complete-registration/components/complete-registration-form/complete-registration-form.tsx]:
```suggestion
import { AuthRoute } from 'src/common/enums';
import { useAppDispatch, useAppNavigation } from 'src/hooks';
```

File: src/screens/complete-registration/components/complete-registration-form/complete-registration-form.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+      dispatch(authActions.verifyToken(accessToken))
+        .unwrap()
+        .then((success: { token: string }) => {
+          dispatch(
+            authActions.completeRegistration({
+              password: values.password,
+              token: success.token,
+            })
+          );
+        });
```
Comment (liza-veis) [inline src/screens/complete-registration/components/complete-registration-form/complete-registration-form.tsx]: Move all this logic to one action

File: src/screens/complete-registration/components/complete-registration-form/complete-registration-form.tsx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+            />
+            <FormPasswordInput
+              name="repeat_password"
+              style={styles.formField}
+              label="Repeat Password"
+            />
+            <MainButton
+              style={styles.button}
+              onPress={handleSubmit}
+              disabled={!isValid && accessToken !== undefined}
```
Comment (liza-veis) [inline src/screens/complete-registration/components/complete-registration-form/complete-registration-form.tsx]:
```suggestion
              disabled={!isValid && accessToken}
```

File: src/services/http/index.ts [hunk 5]
```diff
@@ -63,13 +65,20 @@ class Http {
...
     hasAuth,
+    accessToken,
   }: GetHeadersParams): Promise<Headers> {
     const headers = new Headers();
 
     if (contentType) {
       headers.append(HttpHeader.CONTENT_TYPE, contentType);
     }
 
+    if (accessToken !== null) {
```
Comment (liza-veis) [inline src/services/http/index.ts]:
```suggestion
    if (accessToken) {
```

**Review by liza-veis — CHANGES_REQUESTED** [review]

File: android/app/src/main/AndroidManifest.xml [hunk 6]
```diff
@@ -29,9 +29,9 @@
           <action android:name="android.intent.action.VIEW" />
           <category android:name="android.intent.category.DEFAULT" />
           <category android:name="android.intent.category.BROWSABLE" />
-          <data android:scheme="@strings/deep_linking_scheme" />
-          <data android:scheme="http" android:host="@strings/backend_domain" />
-          <data android:scheme="https" android:host="@strings/backend_domain" />
+          <data android:scheme="@string/app_scheme" />
```
Comment (Liza-Veis) [inline android/app/src/main/AndroidManifest.xml]: Remove changes that is not related to your pr

File: src/components/auth-screen/auth-screen.tsx [hunk 7]
```diff
@@ -8,9 +8,9 @@ import { MainButton, Text } from '..';
 import useStyles from './styles';
 
 type AuthScreenProps = {
-  secondaryMessage: string;
-  secondaryButtonTitle: string;
-  navigateScreen: AuthRoute;
+  secondaryMessage?: string;
+  secondaryButtonTitle?: string;
+  navigateScreen?: AuthRoute;
```
Comment (liza-veis) [inline src/components/auth-screen/auth-screen.tsx]: Don't make them optional

File: src/services/auth-api/index.ts [hunk 8]
```diff
@@ -61,6 +61,33 @@ class AuthApi {
       },
     });
   }
+
+  public async completeRegistration(payload: {
+    token: string;
+    password: string;
+  }) {
+    const authToken: { token: string } = await this.#http.load(
```
Comment (liza-veis) [inline src/services/auth-api/index.ts]:
```suggestion
    const { token: accessToken }: { token: string } = await this.#http.load(
```

File: src/services/http/index.ts [hunk 9]
```diff
@@ -63,13 +65,20 @@ class Http {
...
     }
 
+    if (accessToken) {
+      const token = accessToken;
+
+      headers.append(HttpHeader.AUTHORIZATION, token || '');
+    }
+
     if (hasAuth) {
       const token = await this.#storage.getItem(SecureStorageKey.ACCESS_TOKEN);
```
Comment (liza-veis) [inline src/services/http/index.ts]:
```suggestion
    if (hasAuth) {
      let token = accessToken;
      
      if (!token) {
        token = await this.#storage.getItem(SecureStorageKey.ACCESS_TOKEN);
      }
```

File: src/services/auth-api/index.ts [hunk 10]
```diff
@@ -61,6 +61,33 @@ class AuthApi {
...
+      {
+        method: HttpMethod.PATCH,
+        contentType: ContentType.JSON,
+        payload: JSON.stringify({
+          password: payload.password,
+          firstName: '',
+          lastName: '',
+          position: '',
+        }),
+        hasAuth: false,
```
Comment (liza-veis) [inline src/services/auth-api/index.ts]:
```suggestion
```

**Review by liza-veis — CHANGES_REQUESTED** [review]

**Review by yevhenprybytok — APPROVED** [review]

**Review by liza-veis — APPROVED** [review]

**Pull request merged**